whedon.rb

```diff
@@ -108,12 +115,66 @@ def robawt_respond
   when /\A@whedon assignments/i
     reviewers, editors = assignments
     respond erb :assignments, :locals => { :reviewers => reviewers, :editors => editors, :all_editors => @config.editors }
+  when /\A@whedon generate pdf/i
+    puts "Attempting to compile PDF"
+    respond "```\nPDF at: #{process_pdf}\n```"
   end
 end
 
 # How Whedon talks
-def respond(comment)
-  settings.github.add_comment(@nwo, @issue_id, comment)
+def respond(comment, nwo=nil, issue_id=nil)
+  nwo ||= @nwo
+  issue_id ||= @issue_id
+  settings.github.add_comment(nwo, issue_id, comment)
+end
+
+# Download and compile the PDF
+def process_pdf
+  puts "In #process_pdf"
+  # TODO refactor this so we're not passing so many arguments to the method
+
+  pdf_path = WhedonWorker.new.perform(@config.papers, @config.site_host, @config.site_name, @nwo, @issue_id)
+
+  puts "Creating Git branch"
+  create_or_update_git_branch
+
+  puts "Uploading #{pdf_path}"
+  create_git_pdf(pdf_path)
+  # WhedonWorker.perform_async(@config.papers, @config.site_host, @config.site_name, @nwo, @issue_id)
+end
+
+# GitHub stuff (to be refactored!)
```

Comment:
Working with the GitHub API like this is kinda weird. We have to use the [Repository Contents API](https://developer.github.com/v3/repos/contents/#create-a-file) which is a little weird to work with. Basically we have to:

- First check if the PDF is already present in the repo.
  - If it is, get the SHA of the PDF blob (we need this to delete the old file)
    - Delete the old file
  - If it isn't, create a new branch for the file
- Upload the compiled PDF